src/ripple/conditions/impl/utils.h

```diff
@@ -31,338 +33,195 @@
-{
-    if (begin == end)
-        Throw<std::length_error>("empty buffer");
-
-    std::size_t bytes = *begin++;
+    if (p.length & 0x80)
+    { // Long form length:
+        std::size_t const cnt = p.length & 0x7F;
 
-    if (bytes < 128)
-        return { begin, bytes };
+        if (cnt == 0)
+        {
+            ec = error::generic;
+            return p;
+        }
 
-    bytes &= 0x7F;
+        if (cnt > sizeof(std::size_t))
+        {
+            ec = error::generic;
+            return p;
+        }
 
-    if (bytes > 4)
-        Throw<std::length_error>("overlong encoded length: " + std::to_string(bytes));
+        // Account for the first two bytes in the preamble:
+        if (cnt >= s.size())
+        {
+            ec = error::generic;
+            return p;
+        }
 
-    std::size_t len = 0;
+        p.length = 0;
 
-    if (std::distance (begin, end) < bytes)
-        Throw<std::length_error>("short encoded length: " + std::to_string(bytes));
+        for (std::size_t i = 0; i != cnt; ++i)
+            p.length = (p.length << 8) + s[i];
 
-    while (bytes--)
-        len = (len << 8) | *begin++;
+        s += cnt;
 
-    return { begin, len };
-}
-
-/** Encode a fixed-size octet string: OER 2.6 (2) */
-template <class InputIt, class OutputIt>
-OutputIt
-encode_octetstring(InputIt begin, InputIt end, OutputIt out)
-{
-    while (begin != end)
-        *out++ = *begin++;
+        if (p.length == 0)
+        {
+            ec = error::generic;
+            return p;
+        }
+    }
 
-    return out;
+    ec = {};
+    return p;
 }
 
-/** Encode a dynamic size octet string: OER 2.6 (1) */
 inline
-std::size_t
-predict_octetstring_size(std::size_t size)
+Buffer
+parseOctetString(Slice& s, std::uint32_t count, std::error_code& ec)
 {
-    // Alternatively, always guess 4 + size and call it a day?
-    if (size <= 0x7F)
-        return size + 1;
-
-    // Decide how many bytes we need:
-    if (size <= 0xFFFF)
-        return size + 3;
-
-    if (size <= 0xFFFFFF)
-        return size + 4;
-
-    if (size <= 0xFFFFFFFF)
-        return size + 5;
+    if (count > s.size())
+    {
+        ec = error::buffer_underfull;
+        return {};
+    }
 
-    Throw<std::length_error>("overlong encoding length: " + std::to_string(size));
+    Buffer b(s.data(), count);
+    s += count;
+    ec = {};
+    return b;
 }
 
-/** Encode an dynamic size octet string: OER 2.6 (1) */
-template <class InputIt, class OutputIt>
-OutputIt
-encode_octetstring(std::size_t size, InputIt begin, InputIt end, OutputIt out)
-{
-    // This will encode the length first, followed by the
-    // payload octets:
-    return encode_octetstring (
-        begin,
-        end,
-        oer::encode_length (size, out));
-}
-
-template <class Integer, class OutputIt>
-std::enable_if_t<std::is_unsigned<Integer>::value>
-encode_varuint (Integer value, OutputIt out)
+template <class Integer>
+Integer
+parseInteger(Slice& s, std::size_t count, std::error_code& ec)
```

Comment:
Not sure if @seelabs meant the same thing by overflow, but this should assert or check that count isn't greater than `sizeof(Integer)`.